test/test_allocation_domain.cpp

```diff
@@ -888,4 +889,121 @@ TEST_F(AllocationDomainTest, NHWC2d_To_NHWC2d_cacheAfter_CUDA) {
   testValidate(&fusion, cg_outputs, {t0}, {t0}, __LINE__, __FILE__);
 }
 
+// Check that the maximum vectorizable width is computed correctly
+TEST_F(AllocationDomainTest, NHWCMaxVectorizableWidth_CUDA) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(4, DataType::Half);
+  fusion.addInput(tv0);
+  auto tv1 = set(tv0);
+  fusion.addOutput(tv1);
+
+  std::vector<IterDomain*> tv0_nhwc = {
+      tv0->axis(0), tv0->axis(2), tv0->axis(3), tv0->axis(1)};
+  tv0->setAllocationDomain(tv0_nhwc, false);
+
+  int n = 2, h = 2, w = 2, c = 10001;
+
+  // fully discontiguous
+  {
+    tv0->setContiguity(false);
+    auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+    at::Tensor t0 = at::randn({n * h * w * c}, options)
+                        .as_strided({n, c, h, w}, {5, 2, 3, 4});
+    auto vec = SchedulerRuntimeInfo(fusion_ptr.get(), {t0})
+                   .getMaxVectorizableWidth(tv0);
+    ASSERT_EQ(vec, 1);
+  }
+
+  // c contiguous, bad stride
+  {
+    tv0->setContiguity({false, false, false, true});
+    auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+    at::Tensor t0 = at::randn({n * h * w * c}, options)
+                        .as_strided({n, c, h, w}, {5, 1, 3, 7});
+    auto vec = SchedulerRuntimeInfo(fusion_ptr.get(), {t0})
+                   .getMaxVectorizableWidth(tv0);
+    ASSERT_EQ(vec, 1);
+  }
+
+  // c contiguous, good stride
+  {
+    tv0->setContiguity({false, false, false, true});
+    auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+    at::Tensor t0 = at::randn({n * h * w * c}, options)
+                        .as_strided({n, c, h, w}, {16, 1, 8, 4});
+    auto vec = SchedulerRuntimeInfo(fusion_ptr.get(), {t0})
+                   .getMaxVectorizableWidth(tv0);
+    ASSERT_EQ(vec, 1);
+  }
+
+  // wc contiguous, bad stride
+  {
+    tv0->setContiguity({false, false, true, true});
+    auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+    at::Tensor t0 = at::randn({n * h * w * c}, options)
+                        .as_strided({n, c, h, w}, {5, 1, 3, c});
+    auto vec = SchedulerRuntimeInfo(fusion_ptr.get(), {t0})
+                   .getMaxVectorizableWidth(tv0);
+    ASSERT_EQ(vec, 1);
+  }
+
+  // wc contiguous, good stride
+  {
```

Comment:
Is the generated code vectorized with a factor of 2?

Reply:
No, this fusion is not scheduled. `getMaxVectorizableWidth` is mostly used to compute heuristics and determine the max safe vectorization factor. It's the scheduler's decision to vectorize or not.

Reply:
OK, that was my guess as I haven't seen any update about the scheduler.

Reply:
Schedulers needs a lot more work to do, and I will likely do these work in a bunch of PRs.